quickwit/quickwit-search/src/cluster_client.rs

```diff
@@ -191,7 +191,7 @@ impl ClusterClient {
         client.leaf_list_terms(request.clone()).await
     }
 
-    /// Attempts to store a given search context within the cluster.
+    /// Attempts to store a given key value pair within the cluster.
```

Comment:
i don't think this is ever used for anything else than a search/scroll context, nor should it be (that's not a persistent storage)

Reply:
Agreed. Unfortunately, everything in the code, including core abstractions and RPCs, wants to make this generic. I just tried to add more consistency to that to help the reader. 

I left hints in `quickwit-search/src/scroll_context.rs` and `quickwit-search/src/metrics.rs` about how this store is currently used. I also added some extra clarifications to the function contract [here](https://github.com/quickwit-oss/quickwit/pull/5742/commits/bd570a4162def97de495fbba0bc750cbaba88dac).